cli/src/version.rs

```diff
@@ -79,40 +76,43 @@ fn get_version_disk(
     Ok(local_version)
 }
 
-pub fn get_latest_release() -> Result<Release, Box<dyn Error + 'static>> {
-    let platform: Option<&str> = match OS {
-        "macos" => Some("darwin"),
-        "linux" | "windows" => Some(OS),
-        _ => None,
-    };
+pub(crate) fn get_latest_release(cdn_host: &str) -> Fallible<Release> {
+    let platform: &str = match OS {
+        "macos" => Ok("darwin"),
+        "linux" | "windows" => Ok(OS),
+        _ => Err(ErrorDetails::UnsupportedPlatformError { os: OS.to_string() }),
+    }?;
+
+    let url = format!("{}/cli/{}", cdn_host, platform);
 
-    let platform = platform.ok_or(ErrorDetails::UnsupportedPlatformError {
-        os: String::from(OS),
-    })?;
-    let url = format!("{}/cli/{}", domain(), platform);
     debug!("Fetching latest version from {}", url);
     let resp = reqwest::blocking::Client::new()
         .head(&url)
         .header("User-Agent", "Apollo CLI")
         .send()
-        .map_err(|_err| ExitCode::NetworkError)
-        .unwrap();
+        .map_err(|_err| ErrorDetails::ReleaseFetchError)?;
 
     if !resp.status().is_success() {
         debug!("Request to load the latest release");
```

Comment:
Why keep this around at all and not wrap it into the new `debug` statement below?

Reply:
hm yeah I'll fix this. That line doesn't make much sense to me anyway.